update-component.sh

```
@@ -10,100 +10,163 @@ source common-env
-
-   write-env java_path "$java_path"
-   write-env "java_download_url_current" "$java_url"
-
-   echo
-   echo "Done"
+    local java_url="$1"
+
+    block-print "Downloading java"
+    curlp -o "java.tar.gz" "$java_url"
+    block-print "Extracting java"
+    local java_path="$(tar -tf "java.tar.gz" | head -n 1 | cut -d '/' -f 1)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)